zstacklib/zstacklib/utils/fileUploader.py

```
@@ -0,0 +1,289 @@
+                self._fail_task(task, 'capacity not enough for size: %d' % param.image_size)
+
+        if param.slice_offset + param.slice_size == param.image_size:
+            slice_count = param.slice_index + 1
+        else:
+            slice_count = (param.image_size - 1) / param.slice_size + 1
+
+        if not task.slice_count:
+            task.slice_count = slice_count
+        elif task.slice_count != slice_count:
+            raise Exception(
+                "every upload request for image[uuid:%s] should has the same slice size and image size" % param.image_uuid)
```

Ruff (0.13.1)

Create your own exception

(TRY002)